src/hooks/protections/receive_net_message.cpp

```diff
@@ -71,193 +71,214 @@ namespace big
 					break;
 				}
 			}
-			if (player && get_msg_type(msgType, buffer))
+
+			if (get_msg_type(msgType, buffer))
 			{
-				switch (msgType)
-				{
-				case rage::eNetMessage::MsgTextMessage:
-				case rage::eNetMessage::MsgTextMessage2:
+				if (msgType == rage::eNetMessage::MsgTransitionLaunchNotify)
 				{
-					char message[256];
-					buffer.ReadString(message, 256);
-
-					if (player->is_spammer)
-						return true;
-
-					if (spam::is_text_spam(message))
+					if (frame->m_connection_identifier != gta_util::get_network()->m_transition_session.m_connection_identifier)
 					{
-						if (g.session.log_chat_messages)
-							spam::log_chat(message, player, true);
-						player->is_spammer = true;
-						if (g.session.kick_chat_spammers)
+						if (player)
 						{
-							((player_command*)command::get(RAGE_JOAAT("breakup")))->call(player, {});
+							g_notification_service->push_error("Protections", std::format("Blocked invalid transition launch notify crash from {}", player->get_name()));
 						}
-						return true;
-					}
-					else
-					{
-						if (g.session.log_chat_messages)
-							spam::log_chat(message, player, false);
-
-						if (g.session.chat_commands && message[0] == g.session.chat_command_prefix)
-							command::process(std::string(message + 1), std::make_shared<chat_command_context>(player));
-					}
-					break;
-				}
-				case rage::eNetMessage::MsgScriptMigrateHost:
-				{
-					if (player->m_host_migration_rate_limit.process())
-					{
-						if (player->m_host_migration_rate_limit.exceeded_last_process())
+						else
 						{
-							session::add_infraction(player, Infraction::TRIED_KICK_PLAYER);
-							g_notification_service->push_error("Protections", std::format("{} tried to OOM kick you!", player->get_name()));
+							g_notification_service->push_error("Protections", "Blocked invalid transition launch notify remote crash");
 						}
+
 						return true;
 					}
-					break;
 				}
-				case rage::eNetMessage::MsgRemoveGamersFromSessionCmd:
+
+				if (player)
```

Comment:
nit: invert this if-statement and similar ones above to avoid indentation hell, either by moving this all to a separate function or by using gotos. It hurts to read a switch case that's 3 levels deep 😵